src/test/lit-element_styling_test.ts

```diff
@@ -890,6 +897,89 @@ suite('Static get styles', () => {
 
         document.body.removeChild(element);
       });
+
+  // Test this in Shadow DOM without `adoptedStyleSheets` only since it's easily
+  // detectable in that case. Look explicitly for no ShadyCSS.
+  const testAdoptedStyleSheets =
+      (typeof ShadowRoot === 'function') &&
+      ('adoptedStyleSheets' in window.ShadowRoot.prototype);
+  (testAdoptedStyleSheets ? test : test.skip)(
+      'Can return CSSStyleSheet where adoptedStyleSheets are natively supported',
+      async () => {
+        const sheet = new CSSStyleSheet();
+        sheet.replaceSync('div { border: 4px solid red; }');
+
+        const base = generateElementName();
+        customElements.define(base, class extends LitElement {
+          static styles = sheet;
+
+          render() {
+            return htmlWithStyles`<div></div>`;
+          }
+        });
+
+        const el = document.createElement(base);
+        container.appendChild(el);
+        await (el as LitElement).updateComplete;
+        const div = el.shadowRoot!.querySelector('div')!;
+        assert.equal(
+            getComputedStyle(div).getPropertyValue('border-top-width').trim(),
+            '4px');
+
+        // When the WC polyfills are included, calling .replaceSync is a noop to
+        // our styles as they're already flattened (so expect 4px). Otherwise,
+        // look for the updated value.
+        const flattenDueToPolyfill = (window.ShadyCSS !== undefined && !window.ShadyCSS.nativeShadow);
+        const expectedValue = flattenDueToPolyfill ? '4px' : '2px';
+        sheet.replaceSync('div { border: 2px solid red; }');
+
+        assert.equal(
+            getComputedStyle(div).getPropertyValue('border-top-width').trim(),
+            expectedValue);
+      });
+
+  // Test that when ShadyCSS is enabled with native support for
+  // adoptedStyleSheets, we can return a CSSStyleSheet that will be flattened
+  // and play nice with others.
+  const testShadyCSSWithAdoptedStyleSheetSupport = (typeof ShadowRoot === 'function') &&
+      ('adoptedStyleSheets' in window.ShadowRoot.prototype) &&
```

Comment:
I don't think this test ever runs. When using the Shadow DOM polyfill, `ShadowRoot` does not have an `adoptedStyleSheets` property. Just taking out that check should make this run.

Reply:
Well, that makes the test fail, because I can't construct `CSSStyleSheet` without `adoptedStyleSheets` support.

I've changed the logic a bit:
```js
  const testShadyCSSWithAdoptedStyleSheetSupport =
      (window.ShadowRoot) &&
      ('replace' in CSSStyleSheet.prototype) &&
      (window.ShadyCSS !== undefined && !window.ShadyCSS.nativeShadow);
```

This uses `ShadyCSS` the same way we do in css-tag.ts, and basically says whether we're flattening the stylesheets because the polyfill is included.